### PR TITLE
panzer cleanup

### PR DESCRIPTION
**Pull request by rppawlo**
Some cleanup of panzer objects.

### CODE REVIEW

File: packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp [hunk 1]
```diff
@@ -95,55 +95,56 @@ namespace panzer {
...
-              vals[col] = unweightedBasis(cell,row,qp) * weightedBasis(cell,col,qp);
+      for (const auto& workset : *worksets) {
 
-            M.sumIntoValues(lid,cLIDs,numIds,vals,true,true);
+        const auto& basisValues = workset.getBasisValues(targetBasisDescriptor_,integrationDescriptor_);
+        
+        const auto& unweightedBasis = basisValues.basis_scalar;
+        const auto& weightedBasis = basisValues.weighted_basis_scalar;
+        
+        // Offsets (this assumes UVM, need to fix)
```
Comment (jmgate) [inline packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp]: Is there an issue to capture the need to fix this at some later point?

Comment (rppawlo) [reply packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp]: Nope.  Will need to add a general ticket to remove UVM use in Pazer. Panzer assumes UVM in a ton of places. As I go over code, I have been adding this comment to flag sections as a reminder. Removing all UVM will be a significant undertaking. Note this code is not actually new. I wrapped this whole function in a new loop over worksets, resulting in an indentation change. The diff tool couldn't figure this out and flagged the whole section as changed.

Comment (jmgate) [reply packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp]: Gotcha.  Sounds good to me.

**Comment by mhoemmen** [comment]
@rppawlo Sorry to miss this one!

**Comment by rppawlo** [comment]
@mhoemmen - no worries - just wanted to keep you in the loop since you ping me on this in #2956

**Pull request merged**